src/ray/raylet/node_manager.cc

```diff
@@ -2102,6 +2129,7 @@ void NodeManager::ResubmitTask(const Task &task) {
     RAY_CHECK_OK(gcs_client_->error_table().PushErrorToDriver(
         task.GetTaskSpecification().JobId(), type, error_message.str(),
         current_time_ms()));
+    TreatTaskAsFailed(task, ErrorType::OBJECT_UNRECONSTRUCTABLE, &required_object_id);
```

Comment:
How about we leave `TreatTaskAsFailed` unchanged and just call the new `MarkObjectsAsFailed` (and maybe rename that to `MarkObjectAsFailed`) here?

Reply:
You normally want to write to all the output ids if possible, right?

Reply:
It's already a driver for sure at this point, and drivers never have return IDs.

Reply:
Ah good point.